packages/strapi-plugin-users-permissions/admin/src/components/Permissions/PermissionRow/index.js

```diff
@@ -0,0 +1,80 @@
+import React, { useMemo } from 'react';
+import { Flex, Text } from '@buffetjs/core';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+
+import getTrad from '../../../utils/getTrad';
+import SubCategory from './SubCategory';
+import RowStyle from './RowStyle';
+import PermissionsWrapper from './PermissionsWrapper';
+
+const PermissionRow = ({ openedPlugin, onOpenPlugin, permissions, isWhite, permissionType }) => {
+  const { formatMessage } = useIntl();
+
+  const subCategories = useMemo(() => {
+    return Object.values(permissions.controllers).reduce((acc, curr, index) => {
```

Comment:
It would be great to have some small tests so we can more easily understand the data your are manipulating. What do you think?